### PR TITLE
Fixed MTY_Frame style and calculation issues

### PR DESCRIPTION
**Pull request by Kodikuu**
This appears to have FAR better behaviour than before.

The main thing is using the correct window style; `WS_OVERLAPPEDWINDOW` is incorrect, but `WS_OVERLAPPED` cannot be used despite being correct, so we should give `WS_CAPTION`, which is equivalent in sizing to `WS_OVERLAPPED`

With the correct style, the weird maths that was used to try and 'correct' it is no longer needed

This fixes the first opening of an application being in the wrong position, potentially off-screen, at high scale factors.
This also fixes being unable to set a specific window size (e.g; 2560x1440 was resulting in 2560x1421)

### CODE REVIEW

**Review by Namaneo — APPROVED** [review]

**Comment by RonaldH** [comment]
We're changing from `WS_OVERLAPPEDWINDOW` which has implied
- WS_OVERLAPPED - The window is an overlapped window. An overlapped window has a title bar and a border.
- WS_CAPTION - The window has a title bar (includes the WS_BORDER style). - WS_BORDER - The window has a thin-line border
- WS_SYSMENU - The window has a window menu on its title bar. 
- WS_THICKFRAME - The window has a sizing border
- WS_MINIMIZEBOX - The window has a minimize button
- WS_MAXIMIZEBOX - The window has a maximize button

To:
- WS_CAPTION - The window has a title bar (includes the WS_BORDER style).
- WS_BORDER - The window has a thin-line border

Where WS_CAPTION automatically implies WS_BORDER
So my questions are:
1) Is WS_BORDER needed, since it's implied
2) Are the minimize/maximize buttons still there, is anything else missing?
3) Did you try the WS_OVERLAPPEDWINDOW style, and removed combinations to see which windows style is the cause?

File: src/windows/appw.c [hunk 1]
```diff
@@ -193,10 +193,10 @@ static void app_register_raw_input(USHORT usage_page, USHORT usage, DWORD flags,
 static void app_adjust_window_rect(MTY_App *app, float scale, RECT *r)
 {
 	if (app->AdjustWindowRectExForDpi) {
-		app->AdjustWindowRectExForDpi(r, WS_OVERLAPPEDWINDOW, FALSE, 0, lrint(scale * 96));
+		app->AdjustWindowRectExForDpi(r, WS_CAPTION | WS_BORDER, FALSE, 0, lrint(scale * 96));
 
 	} else {
-		AdjustWindowRect(r, WS_OVERLAPPEDWINDOW, FALSE);
+		AdjustWindowRect(r, WS_CAPTION | WS_BORDER, FALSE);
```
Comment (RonaldH-Parsec) [inline src/windows/appw.c]: So, we're changing from:
WS_OVERLAPPEDWINDOW = WS_OVERLAPPED | WS_CAPTION | WS_SYSMENU | WS_THICKFRAME | WS_MINIMIZEBOX | WS_MAXIMIZEBOX
With implied WS_BORDER by WS_CAPTION

To:
WS_CAPTION | WS_BORDER

So, Minimizebox, and Maximizebox are now hidden? Did you check if removing one of the flags fixes it? Does everything else still work properly?

Comment (Kodikuu) [reply src/windows/appw.c]: WS_CAPTION alone is enough for this. This does not actually interact with the window, so all we're trying to emulate here is the modifiers to the _dimensions_ of the window. 

The minimise and maximise buttons do not affect the window size

File: src/windows/appw.c [hunk 2]
```diff
@@ -183,10 +183,10 @@ static void app_register_raw_input(USHORT usage_page, USHORT usage, DWORD flags,
 static void app_adjust_window_rect(MTY_App *app, float scale, RECT *r)
 {
 	if (app->AdjustWindowRectExForDpi) {
-		app->AdjustWindowRectExForDpi(r, WS_OVERLAPPEDWINDOW, FALSE, 0, lrint(scale * 96));
+		app->AdjustWindowRectExForDpi(r, WS_CAPTION, FALSE, 0, lrint(scale * 96));
 
 	} else {
-		AdjustWindowRect(r, WS_OVERLAPPEDWINDOW, FALSE);
+		AdjustWindowRect(r, WS_CAPTION, FALSE);
```
Comment (dvijayak) [inline src/windows/appw.c:190]: Just looked at what WS_CAPTION does compared to WS_OVERLAPPEDWINDOW. It appears to be the case that with this change we're throwing away the min/max buttons as well as the system context menu when you right-click on the title bar. Is this intentional? I don't see how it's relevant to the fix.

**Review by dvijayak — CHANGES_REQUESTED** [review]
Looks like a nice, elegant fix, my comment notwithstanding.

**Comment by Kodikuu** [comment]
It's not relevant to the fix, which is why they're not included.
OverlappedWindow also includes a thick border, which is the problem.

WS_CAPTION is the minimal necessary to represent the window properly in
terms of sizing

File: src/windows/appw.c (hunk 2 above)
Comment (Kodikuu) [reply src/windows/appw.c:190]: [Window styles](https://learn.microsoft.com/en-us/windows/win32/winmsg/window-styles)

WS_OVERLAPPEDWINDOW;
- WS_OVERLAPPED (correct, but not allowed for AdjustWindowRect)
- WS_CAPTION (correct and allowed)
- WS_SYSMENU (not necessary for determining size)
- WS_THICKFRAME (We do not use a thick frame)
- WS_MINIMIZEBOX (not necessary for determining size)
- WS_MAXIMIZEBOX (not necessary for determining size)

WS_CAPTION is the minimal set of flags necessary to define the correct window sizing

Comment (dvijayak) [reply src/windows/appw.c:190]: Ah I get it, it's just a query for size, not _actually_ changing the window style.

**Review by dvijayak — APPROVED** [review]

**Pull request closed without merging**